# Bugs
[ ] Release Resources when __call__ finishes. As the object maybe allocate expensive stuff in the constructor, it can 
[ ] Do not kill previous computations on exception, but behave the same as it would be a single core loop

# Features
[ ] Run on cluster